creusot/src/translation/pearlite.rs

```diff
@@ -164,6 +164,35 @@ struct ThirTerm<'a, 'tcx> {
 // TODO: Ensure that types are correct during this translation, in particular
 // - Box, & and &mut
 impl<'a, 'tcx> ThirTerm<'a, 'tcx> {
+    fn body_term(&self, expr: ExprId) -> CreusotResult<Term<'tcx>> {
+        let body = self.expr_term(expr)?;
+        let owner_id = util::closure_owner(self.tcx, self.item_id.into());
```

Comment:
I'm a worried this won't play well with Pearlite closures, where we wouldn't want to look at the closure owner for params but the closure itself.

Reply:
Additionnally, what about when we are looking at the contracts of a program closure?

Reply:
Perhaps, we could make `ensures` and `requires` duplicate the parent's signature in the generated closure so we don't need to look at the parents to extract the params?

Reply:
My first thought would be to get the parent if the def_id had the attribute `#[creusot::decl::spec]`, but looking at the macro expansion it seems like the specs on closures are siblings of the closure, not children.  Given an arbitrary spec_closure, it also looks like there is no good way to tell if it specifies the outer function or a different closure so I would know whether or not to use the parent.

Reply:
I'd prefer not to touch creusot_contracts_proc, but that doesn't seem possible.  I'm trying to decide if it would be better to duplicate the arguments as you suggested or go in the other direction and push spec_closures into the closures they specify and remove their arguments as well.  If we did that would it be possible to do away with IDs (`#[creusot::item]`) since all specifications would be directly contained in the thing they specify?

Reply:
I noticed you were using `dummy_call` and `closure_result` to help with type inference, if the spec_closure no longer repeats the arguments `dummy_call` won't be needed.  We can also modify the closure being specified to call `pub fn closure_result<R, F: FnOnce(R) -> bool>(_: F, _: R)` with it's ensures clause and return value instead of doing it in the spec closure.  Is there any other reason why this approach wouldn't work?

Reply:
>  go in the other direction and push spec_closures into the closures they specify and remove their arguments as well

I think the reason I did it that way is that we break type inference otherwise. We want to let real closure guide inference as much as possible. A possibly a little crazy idea I just had was: push the spec_closures in the program closure but put it at the *end* of the program closure, so hopefully type inference infers the arg types before getting to the specification.

Reply:
I is sort of what I was thinking for the `return` arg.  I was wondering if we could do away with the other args all together like we do with top level specs.